reactive/etcd.py

```diff
@@ -175,58 +145,190 @@ def install_etcd():
 
 
 @when('etcd.installed')
-@when_not('etcd.configured')
-def configure_etcd():
-    etcd_helper = EtcdHelper()
-    cluster_data = {'private_address': unit_get('private-address')}
-    cluster_data['unit_name'] = etcd_helper.unit_name
-    cluster_data['management_port'] = config('management_port')
-    cluster_data['port'] = config('port')
-    # The leader broadcasts the cluster settings, as the leader controls the
-    # state of the cluster. This assumes the leader is always initializing new
-    # clusters, and may need to be adapted later to support existing cluster
-    # states.
-    if is_leader():
-        status_set('maintenance', "I am the leader, configuring single node")
-        cluster_data['token'] = etcd_helper.cluster_token()
-        cluster_data['cluster_state'] = 'existing'
-        cluster_data['cluster'] = etcd_helper.cluster_string()
-        cluster_data['leader_address'] = unit_get('private-address')
-        # Actually broadcast that gnarly dictionary
-        leader_set(cluster_data)
-        # leader assumes new? seems to work.
-        cluster_data['cluster_state'] = 'new'
-    else:
-        status_set('maintenance', 'registering unit with etcd-leader')
-        cluster_data['token'] = leader_get('token')
-        cluster_data['cluster_state'] = leader_get('cluster_state')
-        cluster_data['cluster'] = etcd_helper.cluster_string()
-        cluster_data['leader_address'] = leader_get('leader_address')
-        # self registration provided via the helper class
-        etcd_helper.register(cluster_data)
+@when('etcd.ssl.placed')
+@when_not('leadership.is_leader')
+@when_not('etcd.registered')
+def register_node_with_leader():
+    '''
+    Control flow mechanism to perform self registration with the leader.
+
+    Before executing self registration, we must adhere to the nature of offline
+    static turnup rules. If we find a GUID in the member list without peering
+    information the unit will enter a race condition and must wait for a clean
+    status output before we can progress to self registration.
+    '''
+    # We're going to communicate with the leader, and we need our bootstrap
+    # startup string once.. TBD after that.
+    etcdctl = EtcdCtl()
+    bag = EtcdDatabag()
+    # Assume a hiccup during registration and attempt a retry
+    if bag.cluster_unit_id:
```

Comment:
It is not clear to me how cluster_unit_id is being set or not in the initalizer. It seems to be an important variable in this method. Extra comments here would go far to help me understand how EtcdDatabag could either set or not set cluster_unit_id.